### ISSUE TITLE
Please Support for Apple M1 pro/max

### ISSUE THREAD

**Issue by OhWhoYouAre** [deepfakes/faceswap#1197]
As we know, Apple release New silicon named M1 pro/max.

It has  powerful GPUs and CPUs.

Is there any chance to run FaceSwap on new Mac book pro?

**Comment by zhuzeyu22** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by bryanlyon** [comment]
Like Torzdf said in https://github.com/deepfakes/faceswap/issues/1143#issuecomment-871276880 We cannot support what we don't have access to.  If anyone wishes to donate an M1 device, we might be able to support it but we cannot make any promises.

**Comment by geewiz94** [comment]
You could try my fork. I only managed to get CPU training working (but no extract/convert and GPU training doesn't seem to work):
https://github.com/geewiz94/faceswap

Some notes on my progress in this thread, but for now I'm mostly waiting for Tensorflow 2.7 or Python 3.9 support:
https://forum.faceswap.dev/viewtopic.php?f=16&t=1830

**Comment by daniellivingston** [comment]
@geewiz94 I think we've been replicating each other's work. Here is my fork: https://github.com/deepfakes/faceswap/compare/master...daniellivingston:dev/livingston

I have GPU support working (verified w/ system monitor). I don't know how stable things are at the moment, but I am able to extract and train.

**Comment by geewiz94** [comment]
@daniellivingston Very nice! I tried your fork and can confirm that extract/train and also conversion is working! I think we need to check which dependencies are actually needed for installation on a fresh system. But this is great for running some benchmarks and tests!

Spotted an instance where pynvx is still being used, but that's a quick fix:
https://github.com/daniellivingston/faceswap/blob/b76879af13bba54afbbe797d8ac5d12398820e27/lib/gpu_stats.py#L345-L346

**Comment by geewiz94** [comment]
@daniellivingston I did some training tests with the various models: All of them seem to work, except for Phaze-A with the Stojo preset, where the images turn completely red after a few iterations. I think this is due to the selected Encoder Architecture not being fs_original. I tried changing this line:
https://github.com/daniellivingston/faceswap/blob/b76879af13bba54afbbe797d8ac5d12398820e27/plugins/train/model/phaze_a.py#L8
to `from tensorflow.keras import applications as kapp`.
Training then runs but doesn't seem to be going anywhere (again, red images).
